### PR TITLE
feat: add Intel XPU (Arc/Battlemage) support

### PR DESCRIPTION
**Pull request by kahlun**
## Intel XPU (Arc/Battlemage) Support

Enables VeOmni training on Intel XPU (Arc Pro B60 / Battlemage GPU) with XCCL collective backend.

### What Changed

This PR adds 11 targeted patches across 11 files to make VeOmni functional on Intel XPU hardware.

#### Core Device Patches (committed in 5bdc762)

| Patch | File | Change |
|-------|------|--------|
| B1 | `veomni/utils/device.py` | `get_device_type()` detects XPU via `torch.xpu.is_available()` |
| B2 | `veomni/utils/device.py` | `get_dist_comm_backend()` returns `"xccl"` for XPU |
| B3 | `veomni/ops/group_gemm/utils/device.py` | Guard `cuda.get_device_capability()` crash when CUDA absent |
| B5 | `veomni/distributed/torch_parallelize.py` | Accept `"xpu"` as valid `init_device` in weight loading |
| B6 | `veomni/distributed/moe/moe_layer.py` | Guard CUDA-only `group_gemm` import; `xpu` falls back to eager |
| B7 | `veomni/distributed/torch_parallelize.py` | Use `src_data_rank=None` to avoid redundant scatter on XPU |

#### Code Review Fix (61a499e)
- `veomni/distributed/torch_parallelize.py`: Remove redundant `import functools`; use existing `partial` import

#### Test Infrastructure + Data Pipeline (23b33b6)
- `veomni/data/multimodal/video_utils.py`: Wrap `torchcodec` import in try/except — CUDA-only package absent on XPU
- `tests/tools/training_utils.py`: `materialize_weights()` uses `init_device="cpu"` to avoid XPU OOM on large test fixtures

#### Argparse Literal Type (de3aa58 / ce6b6fc)
- `veomni/arguments/arguments_types.py`:
  - Add `"xpu"` to `init_device` Literal type (B8) — without this, `--train.init_device=xpu` fails argparse validation
  - On XPU, auto-downgrade `flash_attention_{2,3,4}` → `sdpa` (flash_attn is CUDA-only)
  - On XPU, auto-downgrade `moe_implementation=fused/fused_quack` → `eager` (fused MoE kernels are CUDA-only)

#### FLA Hybrid-Layer Fallbacks (56cff3e / 4e0e01e)
Qwen3.5 and Qwen3.5-MoE have hybrid linear-attention layers that use FLA (flash-linear-attention) kernels unavailable on XPU:
- `veomni/models/transformers/qwen3_5_moe/generated/patched_modeling_qwen3_5_moe_gpu.py`: F.conv1d fallback for causal_conv1d; torch fallback call for chunk_gated_delta_rule (no cu_seqlens)
- `veomni/models/transformers/qwen3_5/generated/patched_modeling_qwen3_5_gpu.py`: Same fallbacks
- `veomni/models/transformers/qwen3_5/qwen3_5_gpu_patch_gen_config.py`: Same fallbacks in the source template

#### Memory Stats (f313a9f)
- `veomni/utils/helper.py`: Use `.get("num_alloc_retries", 0)` — XPU `memory_stats()` doesn't expose this CUDA-specific key

### Testing

Validated on Intel Arc Pro B60 (Battlemage-G21, 22.7 GB) with `ZE_AFFINITY_MASK=0,1` (2 XPU tiles):

```bash
ZE_AFFINITY_MASK=0,1 CCL_ATL_SHM=1 CCL_BUFFER_CACHE=0 PYTHONPATH=/path/to/veomni \
  python3 -m pytest tests/distributed/test_fsdp_equivalence.py::test_text_fsdp_equivalence[qwen3_5_moe] -v -s
```

Test runs 2 training steps on 1-GPU DDP and 2-GPU FSDP2, then asserts grad_norm equivalence.

### Notes
- FLA hybrid layers (causal_conv1d, chunk_gated_delta_rule) fall back to pure-PyTorch implementations on XPU.  With dyn_bsz (packed sequences), cross-sequence boundary mixing in conv/attention is negligible (kernel_size=4) and does not affect FSDP equivalence.  For strict per-sequence accuracy with dyn_bsz disabled, install flash-linear-attention (CUDA-only requirement).
- Fused attention (flash_attn) and fused MoE are auto-disabled on XPU; SDPA + eager MoE are used instead.

### CODE REVIEW

File: veomni/distributed/torch_parallelize.py [hunk 1]
```diff
@@ -566,7 +566,14 @@ def parallelize_model_fsdp2(
             rank0_load_and_broadcast_weights(model, weights_path, get_device_type(), dtensor_factory=distribute_tensor)
         else:
             logger.info_rank0("Every rank would read weights from disk and expect this to be slow!")
-            load_model_weights(model, weights_path, get_device_type(), dtensor_factory=distribute_tensor)
+            import functools
```
[the remark this answers is not in the document]
Comment (kahlun) [reply veomni/distributed/torch_parallelize.py]: Fixed in 61a499e — removed the redundant `import functools` and switched to the already-imported `partial` directly.

File: veomni/distributed/torch_parallelize.py [hunk 2]
```diff
@@ -566,7 +566,14 @@ def parallelize_model_fsdp2(
...
+            # B7 fix: every rank already has the full checkpoint (see log above), so
+            # the default src_data_rank=0 triggers a redundant dist.scatter() that
+            # wastes bandwidth on CUDA and causes a driver D-state hang on XPU PCIe
+            # (Level Zero IPC not supported). src_data_rank=None → local split only,
+            # zero collective. Bit-identical losses verified on A100 across 5 models.
+            _dt_no_scatter = functools.partial(distribute_tensor, src_data_rank=None)
```
[the remark this answers is not in the document]
Comment (kahlun) [reply veomni/distributed/torch_parallelize.py]: Fixed in 61a499e — now using `partial(distribute_tensor, src_data_rank=None)` as suggested.

**Comment by kahlun** [comment]
to be close/ to more clear about the difference of verl and veomni.
due to used lesser patch/modification can run with verl + veomni backend.
